sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -4301,8 +4391,8 @@ enqueueMemCopyHelper(pi_command_type CommandType, pi_queue Queue, void *Dst,
   // Get a new command list to be used on this call
   ze_command_list_handle_t ZeCommandList = nullptr;
   ze_fence_handle_t ZeFence = nullptr;
-  if (auto Res = Queue->Context->getAvailableCommandList(Queue, &ZeCommandList,
-                                                         &ZeFence))
+  if (auto Res = Queue->Context->getAvailableCommandList(
+          Queue, &ZeCommandList, &ZeFence, !PreferCopyEngine))
```

Comment:
Why do you negate the "PreferCopyEngine" here and in multiple other places too?

Reply:
AH..This is a logical error. I was replacing IsDeviceLocalCopy which would mean 'do not prefer copy engine'. I will correct this. Thanks for catching this. 
Also, currently, I map copy operations to 'compute' engines when we do not have enough data to say if it is a device-device copy. DO we want to keep it this way or do we want  to map them to 'copy' engines?